classes/observer.php

```diff
@@ -0,0 +1,57 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * mod_hotquestion file observers.
+ *
+ * @package    mod_hotquestion
+ * @copyright  2023 mmohohlo-marumo <>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+class mod_hotquestion_observer {
+
+    public static function hotquestion_round(\local_getsmarter\event\module_release $event) {
+        global $DB;
+
+        try {
+            $sql = "
+            SELECT hr.*, c.startdate
+              FROM {hotquestion_rounds} hr
+              JOIN {hotquestion} h ON h.id = hr.hotquestion
+              JOIN {course} c ON c.id = h.course
+              JOIN {course_sections} cs ON cs.course = c.id
+             WHERE hr.starttime > UNIX_TIMESTAMP(DATE_ADD(NOW(), INTERVAL 7 DAY))
+               AND endtime = 0 AND c.enddate = 0
+               AND h.course = ? AND cs.id = ?
+               ";
+
+            $param = array($event->courseid, $event->objectid);
+
+            $questionround = $DB->get_records_sql($sql, $param);
+
+            if (!empty($questionround)) {
+                foreach ($questionround as $round) {
+                    $round->starttime = $round->startdate;
+                    // Update the starttime with the course startdate.
+                    $DB->update_record("hotquestion_rounds", $round, array('id' => $round->id));
```

Comment:
Incorrect use of the update_record function. 3rd param is if it is a bulk update https://docs.moodle.org/dev/Data_manipulation_API#Updating_records